### PR TITLE
Backport 1706

### PR DESCRIPTION
**Pull request by liam923**
Backport [upstream PR 1706](https://github.com/ocaml/merlin/pull/1706), which adds the `syntax-document` command

### CODE REVIEW

File: src/analysis/syntax_doc.ml [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        {
+          name = "Recursive module";
+          description =
+            "A simultaneous definition of modules that can refer recursively \
+             to each others.";
+          documentation = syntax_doc_url "recursivemodules.html";
+        }
+  | (_, Expression _)
+    :: (_, Expression _)
+    :: (_, Expression _)
```
Comment (goldfirere) [inline src/analysis/syntax_doc.ml:193]: This has a `Case` node upstream. Why the change?

**Review by goldfirere — APPROVED** [review]

File: src/analysis/syntax_doc.ml (hunk 1 above)
Comment (liam923) [reply src/analysis/syntax_doc.ml:193]: There was [another commit](https://github.com/ocaml/merlin/commit/fb180ed04dbdd1cdc9157f34af905a75c748ecbd) on the upstream to remove the `Case` node when porting to OCaml 5. I believe the reason for this change is n-ary functions, but I'm not 100% confident for that being the reason. Either way, this change was needed to make the tests pass.
